packages/compile-solidity/src/run.js

```diff
@@ -214,14 +218,18 @@ function prepareOutputSelection({ targets = [] }) {
 /**
  * Load solc and perform compilation
  */
-async function invokeCompiler({ compilerInput, options }) {
+async function invokeCompiler({ compilerInput, options, solc }) {
   const supplierOptions = {
     parser: options.parser,
     events: options.events,
     solcConfig: options.compilers.solc
   };
-  const supplier = new CompilerSupplier(supplierOptions);
-  const { solc } = await supplier.load();
+
+  if (!solc) {
```

Comment:
You'll see here that I've made this `solc` param optional, but maybe we shouldn't worry about breaking the interface to `run()`, since it's entirely internal. I did it this way just to be safe.

Reply:
Yeah, there's no need to worry about breaking the interface to `run`... you may recall I also did so just recently!